convert_wide_to_long_format/convert_wide_to_long_format.py

```diff
@@ -0,0 +1,74 @@
+import os
+import argparse
+import gzip
+import re
+
+COLUMN= 'COLUMN';
+VALUE= 'VALUE';
+
+def map_row_with_header_index(row, header_index, output_file):
+    for header_row in header_index:
+        new_row_dict = {
+            key: row[idx] if isinstance(idx, int) and 0 <= idx < len(row) else idx
```

Comment:
should error out if a row does not have correct amount of columns!! This function probably not needed as writing out is pretty simple in the main function if just collecting the elements in suggested order in comment below